mirrord-layer/src/lib.rs

```diff
@@ -143,6 +144,11 @@ async fn handle_hook_message(
                 });
         }
     }
+    loop {
```

Comment:
This needs to be in the timeout section of the select, also we need that when it occurs we check if last ping was pongd or not